python/tvm/tir/buffer.py

```diff
@@ -141,6 +141,57 @@ def vstore(self, begin, value):
         begin = (begin,) if isinstance(begin, (int, PrimExpr)) else begin
         return _ffi_api.BufferVStore(self, begin, value)  # type: ignore
 
+    def load(self, indices, predicate=None):
```

Comment:
These look very similar to the existing `vload` and `vstore` methods.  Should they be extended instead with a `predicate` argument instead, rather than introducing new methods?

Reply:
We did look at reusing these but decided against it since the vload/vstore api doesn't let you specify the "indices" argument of the underlying buffer load/store which can limit expressing more complex (e.g. non-consecutive) accesses

Reply:
Ah, I see.  I think this is an unfortunate confusion resulting from the name `begin` for the parameter in `vload` and `vstore`.  This parameter would be better named `indices`, because it is used to set the `indices` argument of the buffer load/store.

While there's some additional logic to handle automatic promotion from a scalar index to a vector index when needed ([link](https://github.com/apache/tvm/blob/main/src/tir/ir/buffer.cc#L410)), a vector index can also be provided directly.